tidy3d/plugins/adjoint/components/data/data_array.py

```diff
@@ -312,30 +313,76 @@ def isel(self, **isel_kwargs) -> JaxDataArray:
 
         return self_sel
 
-    def sel(self, indexers: dict = None, method: str = "nearest", **sel_kwargs) -> JaxDataArray:
-        """Select a value from the :class:`.JaxDataArray` by indexing into coordinate values."""
+    def sel(
+        self, indexers: dict = None, method: Literal[None, "nearest"] = None, **sel_kwargs
+    ) -> JaxDataArray:
+        """Select a value from the :class:`.JaxDataArray` by indexing into coordinates by value.
+
+        Parameters
+        ----------
+        method : Literal[None, "nearest"] = None
+            Method to use for matching coordinate values:
+
+            - None (default): only exact matches
+            - nearest: use nearest valid index value
+
+        Returns
+        -------
+        JaxDataArray
+            JaxDataArray with extracted values.
+        """
+        if method not in [None, "nearest"]:
+            raise NotImplementedError(f"Unkown selection method: {method}.")
+
         isel_kwargs = {}
-        for coord_name, sel_kwarg in sel_kwargs.items():
+        for coord_name, vals in sel_kwargs.items():
             coord_list = self.get_coord_list(coord_name)
-            if isinstance(sel_kwarg, (tuple, list, np.ndarray)):
-                sel_kwarg = list(sel_kwarg)
-                isel_kwargs[coord_name] = []
-                for _sel_kwarg in sel_kwarg:
-                    if _sel_kwarg not in coord_list:
-                        raise DataError(
-                            f"Could not select '{coord_name}={_sel_kwarg}', value not found."
-                        )
-                    coord_index = coord_list.index(_sel_kwarg)
-                    isel_kwargs[coord_name].append(coord_index)
-            else:
-                if sel_kwarg not in coord_list:
-                    raise DataError(
-                        f"Could not select '{coord_name}={sel_kwarg}', value not found."
-                    )
-                coord_index = coord_list.index(sel_kwarg)
-                isel_kwargs[coord_name] = coord_index
+
+            try:  # handle non-numeric types (e.g. str)
+                coord_list = jnp.asarray(coord_list)
+            except TypeError:
+                isel_kwargs[coord_name] = self._indices_literal(coord_list, vals)
+                continue
+
+            vals = jnp.atleast_1d(vals)
+            dist = jnp.abs(coord_list[None] - vals[:, None])
+
+            if method is None:
+                indices = jnp.where(jnp.isclose(dist, 0))[1]
+            elif method == "nearest":
+                indices = jnp.argmin(dist, axis=1)
+
+            if indices.size == 0:
+                raise DataError(
+                    f"Could not select '{coord_name}={vals}', some values were not found."
+                )
+
+            isel_kwargs[coord_name] = jnp.squeeze(indices)
```

Comment:
just want to make sure: this `squeeze` doesn't have any unexpected effects if say my values array has a dimension of size 1? I want to make sure we dont "squeeze" out any dimensions in the values that still have a coordinate in `coords`. Can we add a test for this edge case?

Reply:
Oh nice yea good catch, there is a difference between `.sel(x=0)` and `.sel(x=[0])`. Fixed with tests, matches xarray behavior.